lib/gis_robot_suite/structural_file_builder.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module GisRobotSuite
+  # Builds cocina file metadata
+  class StructuralFileBuilder
+    def self.build(*args)
+      new(*args).build
+    end
+
+    # @param [Hash<Symbol,Assembly::ObjectFile>] objects
+    def initialize(assembly_objectfile, access, version)
+      @assembly_objectfile = assembly_objectfile
+      @access = access
+      @version = version
+    end
+
+    attr_reader :assembly_objectfile, :access, :version
+
+    def build
+      Cocina::Models::File.new(file_attributes)
+    end
+
+    private
+
+    def file_attributes
+      {
+        hasMimeType: mimetype,
+        administrative: { publish: true, shelve: true, sdrPreserve: roletype_for_file == 'master' },
```

Comment:
@andrewjbtw We're moving to preserving everything, right?

Reply:
I don't think it would be a problem to preserve everything in this case but there's not general movement on policy. It looks like this is set up to preserve if the role is "master" and not if the role is "derivative" thought I'm having a hard time seeing how that logic is applied. If I'm understanding the code correctly, that logic is explicitly spelled out in the XML-based approach.